agent/src/main/java/io/jenkins/plugins/remotingkafka/Agent.java

```diff
@@ -0,0 +1,92 @@
+package io.jenkins.plugins.remotingkafka;
+
+import hudson.remoting.Command;
+import org.apache.commons.lang3.SerializationUtils;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class Agent {
+    private static final Logger LOGGER = Logger.getLogger(Agent.class.getName());
+
+    private final Options options;
+
+    public Agent(Options options) {
+        this.options = options;
+    }
+
+    public static void main(String... args) throws InterruptedException, IOException {
+        Options options = new Options();
+        Agent agent = new Agent(options);
+
+        CmdLineParser p = new CmdLineParser(options);
+        try {
+            p.parseArgument(args);
+        } catch (CmdLineException e) {
+            LOGGER.log(Level.SEVERE, "CmdLineException occurred during parseArgument", e);
+            p.printUsage(System.out);
+            System.exit(-1);
+        }
+
+        if (options.help) {
+            p.printUsage(System.out);
+            System.exit(0);
+        }
+
+        if (options.name == null) {
+            try {
+                agent.options.name = InetAddress.getLocalHost().getCanonicalHostName();
+            } catch (IOException e) {
+                LOGGER.severe("Failed to lookup the canonical hostname of this agent, please check system settings.");
+                LOGGER.severe("If not possible to resolve please specify a node name using the '-name' option");
+                System.exit(-1);
+            }
+        }
+
+        URL url = new URL(options.master);
+        String consumerTopic = url.getHost() + "-" + url.getPort() + "-" + options.name
+                + KafkaConstants.CONNECT_SUFFIX;
+
+//        // Producer properties test
+//        Properties producerProps = new Properties();
+//        producerProps.put(KafkaConstants.BOOTSTRAP_SERVERS, options.kafkaURL);
+//        producerProps.put(KafkaConstants.KEY_SERIALIZER, "org.apache.kafka.common.serialization.StringSerializer");
+//        producerProps.put(KafkaConstants.VALUE_SERIALIZER, "org.apache.kafka.common.serialization.StringSerializer");
+//        Producer<String, String> producer = new KafkaProducer<>(producerProps);
+//        producer.send(new ProducerRecord<>(consumerTopic, "launch", "test-local"));
+//        LOGGER.info("sent");
```

Comment:
I think it is better to remove unused, commented code. Or if it still provides value, to pull it out somewhere else, such as a well-named method.

This commented-out code is acceptable for this PR, but it should be cleaned up at some point.